### PR TITLE
Fix bug in CSV writer

### PR DESCRIPTION
**Pull request by palvarezc**
Removed unwanted commas that messed up the csv output file.

### CODE REVIEW

File: src/cavendish_particle_tracks/analysis.py [hunk 1]
```diff
@@ -73,7 +73,16 @@ def spoints(self, values):
             point[1] = values[i][1]
 
     def __str__(self):
-        return f"StereoshiftInfo(name={self.name}; sf1={self._sf1}; sf2={self._sf2}; sp1={self._sp1}; sp2={self._sp2}; shift_fiducial={self.shift_fiducial}; shift_point={self.shift_point}; stereoshift={self.stereoshift}; depth_cm={self.depth_cm})"
+        mystring = f"StereoshiftInfo(name={self.name}; "
+        mystring += f"sf1=[{self._sf1[0]} {self._sf1[1]}]; "
+        mystring += f"sf2=[{self._sf2[0]} {self._sf2[1]}]; "
+        mystring += f"sp1=[{self._sp1[0]} {self._sp1[1]}]; "
+        mystring += f"sp2=[{self._sp2[0]} {self._sp2[1]}]; "
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/analysis.py:80]:
```suggestion
        for name, point in zip(["sf1", "sf2", "sp1", "sp2"], [self._sf1, self._sf2, self._sp1, self._sp2]):
            x, y = point
            mystring += f"{name}=[{x} {y}]; "
```

File: src/cavendish_particle_tracks/analysis.py [hunk 2]
```diff
@@ -179,5 +188,11 @@ def calibrate(self) -> None:
     def to_csv(self):
         mystring = ""
         for var in self.vars_to_save():
-            mystring += str(getattr(self, var)) + ","
+            if var in ["rpoints", "dpoints"]:
+                mystring += "["
+                for point in getattr(self, var):
+                    mystring += "[" + str(point[0]) + " " + str(point[1]) + "]; "
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/analysis.py:194]:
```suggestion
                for point in getattr(self, var):
                    x, y = point
                    mystring += "[{x} {y}]; "
```

File: tests/test_save.py [hunk 3]
```diff
@@ -79,3 +80,34 @@ def set_filename_and_close(dialog):
...
             "The file must be a CSV (*.csv) or Pickle (*.pkl) file. Please try again."
         )
+
+
+def test_csv_file_has_correct_columns(cpt_widget, tmp_path, qtbot: QtBot):
+    # start napari and the particle widget, add a single particle
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)  # select the Σ
+    assert len(cpt_widget.data) == 1, "Expecting one particle in the table"
+
+    file_name = "my_file.csv"
```
Comment (samcunliffe) [inline tests/test_save.py:90]:
```suggestion
    file_name = "test_saved_file.csv"
```

File: tests/test_save.py [hunk 4]
```diff
@@ -79,3 +80,34 @@ def set_filename_and_close(dialog):
...
+
+    def set_filename_and_close(dialog):
```
Comment (samcunliffe) [inline tests/test_save.py:92]:
```suggestion
    def set_filename_and_close(dialog):
        # Function of the signature needed to use as a dialog action.
        # Defined internally so we can access the fixtures without passing.
```

**Review by samcunliffe — APPROVED** [review]

File: src/cavendish_particle_tracks/analysis.py [hunk 5]
```diff
@@ -179,5 +189,12 @@ def calibrate(self) -> None:
     def to_csv(self):
         mystring = ""
         for var in self.vars_to_save():
-            mystring += str(getattr(self, var)) + ","
+            if var in ["rpoints", "dpoints"]:
+                mystring += "["
+                for point in getattr(self, var):
+                    x, y = point
+                    mystring += "[{x} {y}]; "
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/analysis.py:196]:
```suggestion
                    mystring += "f[{x} {y}]; "
```

**Review by samcunliffe — CHANGES_REQUESTED** [review]

File: src/cavendish_particle_tracks/analysis.py (hunk 5 above)
Comment (samcunliffe) [reply src/cavendish_particle_tracks/analysis.py:196]: Sorry. Would be good maybe to commit an expected CSV file into test data? Can then do regression testing.

Comment (samcunliffe) [reply src/cavendish_particle_tracks/analysis.py]: Looks like [`filecmp`](https://docs.python.org/3/library/filecmp.html) does it. And it's a builtin.

- https://stackoverflow.com/q/75838502/1444054
   (the first comment about asserting on an empty diff looks nice).

**Review by samcunliffe — APPROVED** [review]

**Comment by samcunliffe** [comment]
@palvarezc I leave you to merge if you're OK with my reimplemented `filecmp`.

**Comment by palvarezc** [comment]
Thanks!

**Pull request merged**